storage/src/mmr/operation.rs

```diff
@@ -81,6 +89,22 @@ impl<K: Array, V: Array> Operation<K, V> {
         }
     }
 
+    /// Create a new floor operation that raises the floor on inactive operations to `loc`. This
+    /// operation is not supported for stores whose operations encode to less than 9 bytes.
+    pub fn floor(loc: u64) -> Self {
+        // Panic if the encoding size is too small to support this operation type.
+        assert!(Self::LEN_ENCODED >= 9);
+        let mut data = Vec::with_capacity(Self::LEN_ENCODED);
+        data.push(Self::FLOOR_CONTEXT);
+        data.extend_from_slice(&loc.to_be_bytes());
+        data.resize(Self::LEN_ENCODED, 0);
+
+        Self {
+            data,
+            _phantom: std::marker::PhantomData,
+        }
+    }
+
     pub fn to_key(&self) -> K {
```

Comment:
If we work directly with the enums and just call `encode` on them when needed (flipping the abstraction), I think this could become clearer (i.e. we should never be able to call to_key/to_value on a FLOOR by construction).

Reply:
they are in array form in the journal though... so if we also keep them around in enum form aren't we basically caching each twice?

Reply:
I think there may be a clean way to do this but let's do it after we get things working.